src/pages/docs/training-certificate/[orderId]/[participantId].tsx

```diff
@@ -0,0 +1,282 @@
+import CTA from '@components/CTA';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useUser } from '@auth0/nextjs-auth0/client';
+import styled from 'styled-components';
+import { useEffect, useRef, useState } from 'react';
+import html2canvas from 'html2canvas';
+import jsPDF from 'jspdf';
+import Image from 'next/image';
+import Logo from '@assets/new/logos/brand-logo.png';
+import TopPaintIcon from '@assets/new/images/top_cert_paint.png';
+import BottomPaintIcon from '@assets/new/images/bottom_cert_paint.png';
+import { TrainingAttendanceType } from '@data/types/trainingTypes/TypeOrthoexTrainingData';
+import { formatDate } from '@utils/index';
+import { toast } from 'react-toastify';
+
+const TrainingCertificate = () => {
+	// @ts-ignore
+	const { user } = useUser();
+	const router = useRouter();
+	const pdfRef = useRef<HTMLDivElement>(null);
+
+	const [attendanceData, setAttendanceData] =
+		useState<TrainingAttendanceType | null>(null);
+	const orderId = router.query.orderId as string;
+	const participantId = router.query.participantId as string;
+
+	useEffect(() => {
+		if (orderId && participantId) {
+			fetch('/api/download-training-certificate', {
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify({ orderId, participantId }),
+			})
+				.then(res => res.json())
+				.then(response => {
+					if (!response.error) {
+						setAttendanceData(response.data);
+					}
+					if (response.error) {
+						toast.error(response.error);
+					}
+				})
+				.catch(err => {
+					console.log(err);
+					toast.error('Training Certificate could not be found');
+				});
+		}
+	}, [orderId, participantId]);
+
+	const downloadPdf = () => {
+		const input = pdfRef.current as HTMLDivElement;
+		html2canvas(input).then(canvas => {
+			const pdf = new jsPDF('portrait', 'mm', 'a4', true);
+			const imgData = canvas.toDataURL('img/png');
+			const pdfWidth = pdf.internal.pageSize.getWidth();
+			const pdfHeight = pdf.internal.pageSize.getHeight();
+			const imgWidth = canvas.width;
+			const imgHeight = canvas.height;
+			const ratio = Math.min(pdfWidth / imgWidth, pdfHeight, imgHeight);
+			const imgX = (pdfWidth - imgWidth * ratio) / 2;
+			const imgY = 30;
+
+			pdf.addImage(
+				imgData,
+				'PNG',
+				imgX,
+				imgY,
+				imgWidth * ratio,
+				imgHeight * ratio,
+			);
+			pdf.save('certificate.pdf');
+		});
+	};
+
+	if (!user)
```

Comment:
look at other authenticated routes, we use `withPageAuthRequired` provided by auth0 to make sure only authenticated users get to the page at all

Reply:
Done ✅✅